uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/bookingrequest/BookingRequestActivity.kt

```diff
@@ -1,97 +1,100 @@
-        BookingRequestViewContract.BookingRequestWidget, LoadingButtonView.Actions, LifecycleObserver {
-
-    private val containerAnimateIn: Animation = AnimationUtils.loadAnimation(context, R.anim.uisdk_slide_in_bottom)
-    private val containerAnimateOut: Animation = AnimationUtils.loadAnimation(context, R.anim.uisdk_slide_out_bottom)
-    private var backgroundFade: TransitionDrawable? = null
+import java.util.*
+
+class BookingRequestActivity : BaseActivity(), BookingRequestContract.View, BookingRequestContract.Actions,
+        BookingPaymentMVP.PaymentViewActions, BookingPaymentMVP.PaymentActions,
+        BookingRequestViewContract.BookingRequestWidget, LoadingButtonView.Actions {
+    override val layout: Int
+        get() = R.layout.uisdk_booking_request
+
     private var isGuest: Boolean = false
 
     private var holdOpenForPaymentFlow = false
 
-    private var presenter: BookingRequestMVP.Presenter = BookingRequestPresenter(this,
-            KarhooUISDK.analytics,
-            KarhooPreferenceStore.getInstance(context.applicationContext),
-            KarhooApi.tripService,
-            KarhooApi.userStore)
+    private lateinit var presenter: BookingRequestContract.Presenter
 
     private val bookingComments: String
         get() = bookingRequestCommentsWidget.getBookingOptionalInfo()
 
-    init {
-        View.inflate(context, R.layout.uisdk_booking_request, this)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        presenter = BookingRequestActivityPresenter(this,
+                KarhooUISDK.analytics,
+                KarhooPreferenceStore.getInstance(this.applicationContext),
+                KarhooApi.tripService,
+                KarhooApi.userStore)
+
         isGuest = isGuest()
         attachListeners()
-        bookingRequestFlightDetailsWidget.setHintText(context.getString(R.string.kh_uisdk_add_flight_details))
+        bookingRequestFlightDetailsWidget.setHintText(this.getString(R.string.kh_uisdk_add_flight_details))
         presenter.setBookingFields(bookingRequestPassengerDetailsWidget.allFieldsValid())
         TextViewCompat.setTextAppearance(bookingRequestLabel, R.style.ButtonText)
+
+        extras?.let { extras ->
+            val quote = extras.getParcelable<Quote>(BOOKING_REQUEST_QUOTE_KEY)
+            val bookingStatus = extras.getParcelable<BookingStatus>(BOOKING_REQUEST_STATUS_KEY)
+
+            if (quote != null) {
+                bookingRequestButton.onLoadingComplete()
+                bookingRequestLayout.visibility = View.VISIBLE
+                presenter.setBookingStatus(bookingStatus)
+                presenter.showBookingRequest(
+                        quote = quote,
+                        outboundTripId = extras.getString(BOOKING_REQUEST_OUTBOUND_TRIP_ID_KEY),
+                        bookingMetadata = extras.getSerializable(BOOKING_REQUEST_METADATA_KEY) as HashMap<String, String>?
```

Comment:
i think if the extras don't have `BOOKING_REQUEST_METADATA_KEY ` it will throw an exception on the map cast

Reply:
Won't it default to a optional (null object)?